### PR TITLE
Server and database

### PR DESCRIPTION
**Pull request by hasansaad1**
relates to #4  set up an Express server and the database and we used the previous functionality of rendering the posts .

### CODE REVIEW

**Comment by Karyum** [comment]
@hasansaad1 let @ionush @SamiSha99 approve it first then me and @claireinez can review

**Review by nj-io — APPROVED** [review]

**Review by SamiSha99 — APPROVED** [review]
Looks good, Approved.

File: database/db_build.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+);
+
+CREATE TABLE posts(
+  post_id SERIAL PRIMARY KEY,
+  user_id INTEGER NOT NULL ,
+  prefferedPayment VARCHAR(100) NOT NULL,
+  country TEXT NOT NULL,
+  city TEXT NOT NULL,
+  coin TEXT NOT NULL,
+  qty INTEGER NOT NULL,
```
Comment (Karyum) [inline database/db_build.sql:23]: im guessing this is quantity?? why not have `quantity` just written there

File: database/db_connection.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const url = require('url');
+require('env2')('config.env');
+
+if (!process.env.DATABASE_URL)
+  throw new Error("enviroment var for database must be set");
+
+const params = url.parse(process.env.DATABASE_URL);
+const [username, password] = params.auth.split(':');
+
+const options = {
```
Comment (Karyum) [inline database/db_connection.js]: you dont need this whole options object just do `connectionString: process.env.DATABASE_URL`

File: database/queries/fetchPosts.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const connect = require('../db_connection');
+
+const posts = (req, res) => {
+  connect.query('SELECT * FROM posts INNER JOIN users ON (posts.user_id = users.user_id)', (err, posts) => {
+    if (err) {
+      cb(err);
```
Comment (Karyum) [inline database/queries/fetchPosts.js]: where is the callback ? i cant see any `cb` in this file

File: database/queries/fetchPosts.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    } else {
+      // console.log('post.rows:', post.rows);
```
Comment (Karyum) [inline database/queries/fetchPosts.js]: delete this line

File: index.html [hunk 5]
```diff
@@ -12,7 +12,7 @@
     <div class="app" id="root">
 
     </div>
-    <script type="text/javascript" src="./public/bundle.js">
+    <script type="text/javascript" src="./bundle.js">
```
Comment (Karyum) [inline index.html:15]: hehehehehe :wink:

File: package.json [hunk 6]
```diff
@@ -4,7 +4,13 @@
   "description": "A cryptocurrency resource website",
   "main": "index.js",
   "dependencies": {
+    "axios": "^0.17.1",
+    "body-parser": "^1.18.2",
+    "env2": "^2.2.0",
+    "eslint-loader": "^1.9.0",
+    "express": "^4.16.2",
     "nodemon": "^1.14.11",
```
Comment (Karyum) [inline package.json]: this should be a dev dep 


(PS: *this probably me lol*)

File: package.json [hunk 7]
```diff
@@ -35,10 +41,12 @@
...
     "lint": "eslint **/*.{js,jsx} --quiet",
-    "build": "webpack",
-    "dev": "webpack-dev-server --hot",
-    "lint": "eslint **/*.{js,jsx} --quiet"
+    "build": "webpack --config webpack.config.js --watch",
+    "dev": "nodemon src/backEnd/server.js",
+    "resetDB": "node ./database/db_build",
+    "start": "node src/backEnd/server.js"
   },
+  "proxy": "http://localhost:8080",
```
Comment (Karyum) [inline package.json]: do you need this after that we need installed express now to server React?

File: src/backEnd/server.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const path = require('path');
+const bodyParser = require('body-parser');
+
+const router = require('./router');
+
+const app = express();
+console.log(path.join(__dirname, "../.."));
```
Comment (Karyum) [inline src/backEnd/server.js]: a whole lots of console logs in this file that were used for debugging so they are not needed  ow :sweat_smile:

File: src/js/PostRendering.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    super(props);
+    this.state = {
+      posts: null
+    };
+  }
+  componentDidMount() {
+    if (!this.state.posts) {
+      axios.get("/api/posts").then(response => {
+        this.setState({posts: response.data});
+        console.log(this.state);
```
Comment (Karyum) [inline src/js/PostRendering.jsx]: a log :x:

Comment (Karyum) [reply src/js/PostRendering.jsx]: also you are not catching the errors! :scream:

File: src/js/Sell.jsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import PostRendering from "./PostRendering";
+import styled from "styled-components";
+
+const SellHeader = styled.h2`
+  text-align: center;
+`;
+const SellSubheader = styled.h3`
+  text-align: center;
+`;
+SellHeader;
```
Comment (Karyum) [inline src/js/Sell.jsx]: what is this doing here?

File: webpack.config.js [hunk 11]
```diff
@@ -4,44 +4,28 @@ const webpack = require('webpack');
...
   stats: {
     colors: true,
-    reasons: true,
     chunks: false
   },
-  plugins: [
-    new webpack.HotModuleReplacementPlugin(),
-    new webpack.NamedModulesPlugin()
-  ],
+  plugins: [],
```
Comment (Karyum) [inline webpack.config.js]: yeah lets delete this line too

**Review by Karyum — CHANGES_REQUESTED** [review]
the changes are pretty small, i love this setup

File: database/db_build.sql (hunk 1 above)
Comment (hasansaad1) [reply database/db_build.sql:23]: we like qty more because we render it.

**Review by Karyum — APPROVED** [review]

**Pull request merged**